server/sun.py

```diff
@@ -17,16 +17,19 @@ def __init__(self, geocoder):
         self._astral = Astral(geocoder=GeocoderWrapper, wrapped=geocoder)
         self._local_time = LocalTime(geocoder)
 
-    def rewrite_cron(self, cron, after, user):
+    def rewrite_cron(self, cron, reference, user):
         """Replaces references to sunrise and sunset in a cron expression."""
 
         # Skip if there is nothing to rewrite.
         if 'sunrise' not in cron and 'sunset' not in cron:
             return cron
 
+        # Replace H:M:S in the reference time so that we get the correct sunrise / set
+        reference_midnight = reference.replace(hour=0, minute=0, second=0, microsecond=0)
```

Comment:
Not sure I understand this one yet. What happens if you don't do this?

Reply:
So the logic to pick the most recent sunrise / sunset seems to depend on the input datetime being midnight for the current day. I checked the output times without the `replace` and they weren't consistent with the implementation that exists in master.

Reply:
Hm, it always takes me a while to wrap my head around this logic...

If I understand this diff correctly, it modifies the meaning of `next_sunrise` and `next_sunset` below to find the *first* sunrise and sunset within the reference day rather than the *next* one after the reference time (which was the previous behavior).

Unless I'm misunderstanding something, this does seem to introduce some issues. For example, when `schedule.delay()` requests the next time to wake up sometime during the day and the entry references `sunset`, then it may get a time in the past.

Reply:
Ahh yeah, you are correct. I missed the interdependence with `delay()`. I will fix that.